analytics/transform.go

```diff
@@ -191,22 +192,116 @@ func (event ParsedEvent) GetValue(field string) (interface{}, error) {
 		return nil, errors.New(fmt.Sprintf("Key %s not a valid atomic field", field))
 	}
 	if event[index] == "" {
-		return nil, errors.New(fmt.Sprintf("Field %s is empty", field))
+		return nil, errors.New(EmptyFieldErr)
 	}
 	kvPairs, err := enrichedEventFieldTypes[index].ParseFunction(enrichedEventFieldTypes[index].Key, event[index])
 	if err != nil {
 		return nil, err
 	}
-	if field == "contexts" || field == "derived_contexts" || field == "unstruct_event" {
-		// TODO: DRY HERE TOO?
-		output := make(map[string]interface{})
+
+	return kvPairs, nil
+}
+
+// GetValue returns the value for a provided atomic field, without processing the rest of the event.
+// For unstruct_event, it returns a map of only the data for the unstruct event.
+func (event ParsedEvent) GetValue(field string) (interface{}, error) {
+	kvPairs, err := event.getParsedValue(field)
+	if err != nil {
+		return nil, err
+	}
+	return kvPairs[0].Value, nil
+}
+
+// GetUnstructEventValue returns the value for a provided atomic field inside an event's unstruct_event field
+func (event ParsedEvent) GetUnstructEventValue(path ...interface{}) (interface{}, error) {
+	field := "unstruct_event"
+
+	kvPairs, err := event.getParsedValue(field)
+	if err != nil {
+		return nil, err
+	}
+
+	// extract the key/value pairs of the event field into a map
+	eventMap := make(map[string]interface{})
```

Comment:
I think this is unnecessary for Unstruct events - this field can only ever have one entry, it's not an array like contexts.

So we can just type switch the `KeyVal.Value` into a `map[string]interface{}` and we've got what we need.